tsdapiclient/tacl2.py

```diff
@@ -63,6 +63,33 @@ def get_guide_options(ctx, args, incomplete):
     return [k for k,v in GUIDES.items() if incomplete in k]
 
 
+def get_dir_contents(ctx, args, incomplete):
+    sep = os.path.sep
+    if os.path.lexists(incomplete):
+        if os.path.isdir(incomplete):
+            if not incomplete.endswith(sep):
+                return [f'{incomplete}{sep}'] # prepare to list
+            else:
+                base, fragment = os.path.dirname(incomplete), os.path.basename(incomplete)
+                return [f'{base}{sep}{entry}' for entry in os.listdir(base) if fragment in entry]
+        elif os.path.isfile(incomplete):
+            base, fragment = os.path.dirname(incomplete), os.path.basename(incomplete)
+            return [incomplete]
+        else:
+            return [] # not sure what this could be (yet)
+    else:
+        if incomplete == '':
+            return os.listdir('.')
+        else:
+            base, fragment = os.path.dirname(incomplete), os.path.basename(incomplete)
+            if base == '' and fragment == '':
+                return [entry for entry in os.listdir('.') if incomplete in entry]
```

Comment:
I think these checks should be changed to `entry.startswith(incomplete)` and `entry.startswith(fragment)` so the autocompletion will behave more like the shell probably would.

Reply:
done